backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -1 +1,53 @@
-export default class TeamEventAttendanceDao {}
+import * as admin from 'firebase-admin';
+import { v4 as uuidv4 } from 'uuid';
+import { memberCollection, teamEventAttendanceCollection } from '../firebase';
+import { DBTeamEventAttendance } from '../types/DataTypes';
+
+export default class TeamEventAttendanceDao {
+  static async createTeamEventAttendance(
+    teamEventAttendance: TeamEventAttendance
+  ): Promise<TeamEventAttendance> {
+    const teamAttendanceRef: DBTeamEventAttendance = {
+      ...teamEventAttendance,
+      member: memberCollection.doc(teamEventAttendance.member.email),
+      ...(typeof teamEventAttendance.hoursAttended === 'number'
+        ? { hoursAttended: teamEventAttendance.hoursAttended }
+        : {}),
+      pending: true,
+      uuid: teamEventAttendance.uuid ? teamEventAttendance.uuid : uuidv4()
+    };
+    await teamEventAttendanceCollection.doc(teamAttendanceRef.uuid).set(teamAttendanceRef);
+    return teamEventAttendance;
```

Comment:
I would return `teamEventAttendance` with the updated fields, not the original object. So `uuid` could potentially change to be the autogenerated one if it was an empty string originally, and `pending` will have been set to true no matter what.